compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -409,11 +409,11 @@ convertTypeDef env x = pure []
 -- here and the `$WFoo` functions generated by GHC. At some point we need to
 -- figure out which ones to keep and which ones to throw away.
 convertCtors :: Env -> Ctors -> ConvertM [Definition]
-convertCtors env (Ctors name isTypeClass tys [o@(Ctor ctor fldNames fldTys)])
+convertCtors env (Ctors name flavour tys [o@(Ctor ctor fldNames fldTys)])
   | isRecordCtor o
   = pure $ [defDataType tconName tys $ DataRecord flds] ++
-      [ defValue name (mkVal $ "$ctor:" <> getOccText ctor, mkTForalls tys $ mkTFuns fldTys (typeConAppToType tcon)) expr
-      | not isTypeClass
+      [ defValue name (mkVal $ "$W" <> getOccText ctor, mkTForalls tys $ mkTFuns fldTys (typeConAppToType tcon)) expr
```

Comment:
Maybe worth pulling `\x -> mkVal $ "$W" <> x` into a top-level function? Having the assumptions about how GHC calls certain functions factored out into a few top-level functions seems much nicer than having this littered through the code.

Reply:
Good point. I'll do it in a cleanup PR.